test/tools/save-logs/save_logs.py

```diff
@@ -0,0 +1,381 @@
+    exit(1)
+
+# Default gather script
+gather_script = "rackhd_gather_info.py"
+
+
+def main():
+    myfile = os.path.basename(__file__)
+    path = os.path.realpath(myfile)
+    cur_dir = os.path.dirname(path)
+
+    ARG_PARSER = argparse.ArgumentParser(description="Command Help")
+    ARG_PARSER.add_argument("-v", action='store_true',
+                            help="Turn on verbose mode")
+    ARG_PARSER.add_argument("-ip", default="0.0.0.0",
+                            help="Appliance IP address or hostname")
+    ARG_PARSER.add_argument("-port", default="",
+                            help="Optional port argument for scp/ssh")
+    ARG_PARSER.add_argument("-user", default="None",
+                            help="Appliance username")
+    ARG_PARSER.add_argument("-pwd", default="None",
+                            help="Appliance password")
+    ARG_PARSER.add_argument("-no_extract", action='store_true',
+                            help="Do not extract the tarball, default to extract")
+    ARG_PARSER.add_argument("-log_path", default="None",
+                            help="Logging Directory Path")
+    ARG_PARSER.add_argument("-target_dir", default="None",
+                            help="Target directory name to store logs")
+
+    # parse arguments to ARGS_LIST dict
+    ARGS_LIST = ARG_PARSER.parse_args()
+    try:
+        CONFIG = json.loads(open("save_logs_conf.json").read())
+        print "save_logs: Using configuration from \"save_logs_conf.json\"."
+        config = True
+    except:
+        print "save_logs: No config file found"
+        config = False
+
+    if ARGS_LIST.user == "None":
+        try:
+            ARGS_LIST.user = CONFIG["username"]
+        except:
+            if config is True:
+                print "Username required"
+                print "Usage: {0} -user <username>".format(myfile)
+            else:
+                print "Missing Required Parameters"
+                print "Usage: ./save_logs.py -ip <ip> -user <username> -pwd <password>"
+            sys.exit(-1)
+
+    if ARGS_LIST.pwd == "None":
+        try:
+            ARGS_LIST.pwd = CONFIG["password"]
+        except:
+            if config is True:
+                print "Password required"
+                print "Usage: {0} -pwd <password>".format(myfile)
+            else:
+                print "Missing Required Parameters"
+                print "Usage: ./save_logs.py -ip <ip> -user <username> -pwd <password>"
+            sys.exit(-1)
+
+    if ARGS_LIST.ip == "0.0.0.0":
+        try:
+            ARGS_LIST.ip = CONFIG["ip"]
+        except:
+            if config is True:
+                print "Stack IP address or hostname required."
+                print "Usage: {0} -ip <ip or hostname>".format(myfile)
+            else:
+                print "Missing Required Parematers"
+                print "Usage: ./save_logs.py -ip <ip> -user <username> -pwd <password>"
+            sys.exit(-1)
+
```

Comment:
The ip address can be a name or ip value, so need some further updates to check that.  For now, the hostname/ip address is verified to be reachable later in the script.  If it is unreachable, the script will exit with an error of unable to access the host.  tested with bunches of bad strings, bad ips, etc.
```
(eh-logub16) hohene@lab17:~/sandbox-github/rackhd/eh-loggather/test/tools/save-logs$ python save_logs.py -user onrack -pwd onrack -ip 1.2.3.4
Saving logs in current directory: /emc/hohene/sandbox-github/rackhd/eh-loggather/test/tools/save-logs
Enter name of target directory, will be created: eh10
Directory: eh10 
ERROR: Unable to access host exiting...
```